### PR TITLE
Small fixes to Dockerfile

### PR DESCRIPTION
**Pull request by paulromano**
This PR has a few small changes to our Dockerfile to ensure it still works:
- Remove `FC` environment variable, which is not needed
- Make Debian frontend non-interactive to avoid prompt asking for timezone
- Explicitly put `--recurse-submodules` on the `git clone` command

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,17 +1,18 @@
 FROM ubuntu:latest
 
 # Setup environment variables for Docker image
-ENV FC=/usr/bin/mpif90 CC=/usr/bin/mpicc CXX=/usr/bin/mpicxx \
+ENV CC=/usr/bin/mpicc CXX=/usr/bin/mpicxx \
     PATH=/opt/openmc/bin:/opt/NJOY2016/build:$PATH \
     LD_LIBRARY_PATH=/opt/openmc/lib:$LD_LIBRARY_PATH \
     OPENMC_CROSS_SECTIONS=/root/nndc_hdf5/cross_sections.xml \
-    OPENMC_ENDF_DATA=/root/endf-b-vii.1
+    OPENMC_ENDF_DATA=/root/endf-b-vii.1 \
```
Comment (AI-Pranto) [inline Dockerfile:8]:
```suggestion
    OPENMC_ENDF_DATA=$HOME/endf-b-vii.1 \
```
We extract tar archive to  `tar -C $HOME -xJ`.

File: Dockerfile [hunk 2]
```diff
@@ -1,17 +1,18 @@
 FROM ubuntu:latest
 
 # Setup environment variables for Docker image
-ENV FC=/usr/bin/mpif90 CC=/usr/bin/mpicc CXX=/usr/bin/mpicxx \
+ENV CC=/usr/bin/mpicc CXX=/usr/bin/mpicxx \
     PATH=/opt/openmc/bin:/opt/NJOY2016/build:$PATH \
     LD_LIBRARY_PATH=/opt/openmc/lib:$LD_LIBRARY_PATH \
     OPENMC_CROSS_SECTIONS=/root/nndc_hdf5/cross_sections.xml \
```
Comment (AI-Pranto) [inline Dockerfile:7]:
```suggestion
    OPENMC_CROSS_SECTIONS=$HOME/nndc_hdf5/cross_sections.xml \
```
Same we extract tar archive to `tar -C $HOME -xJ`

**Review by AI-Pranto — CHANGES_REQUESTED** [review]

**Comment by AI-Pranto** [comment]
@paulromano, Do we really need `ENV CC=/usr/bin/mpicc` this PATH variable?

**Comment by AI-Pranto** [comment]
@paulromano For `ubuntu 20.04` default python is `python 3.8` and also we have `pip3`. So we can replace
```
apt-get install -y python3-pip && \
```
with 
```
RUN apt-get install -y python-is-python3
RUN ln -s /usr/bin/pip3 /usr/bin/pip
```
It's annoying to type `python3` which we can solve by `python-is-python3`.

**Comment by paulromano** [comment]
> @paulromano, Do we really need `ENV CC=/usr/bin/mpicc` this PATH variable?

Probably not actually. Only the C++ compiler is used for compiling code that uses MPI, so we could probably get away with just the `CXX` variable, but it doesn't hurt to set `CC` as well.

**Comment by paulromano** [comment]
@AI-Pranto Thanks for the suggestion about `python-is-python3`. I'll go ahead and add that in. Regarding pip, it looks like in the soon-to-be-released Ubuntu 20.10, `python3-pip` [will install `pip`](https://askubuntu.com/questions/320996/how-to-make-python-program-command-execute-python-3#comment2131288_1031733).

**Comment by paulromano** [comment]
@AI-Pranto It looks like using `$HOME` actually doesn't work in `ENV` commands (it just evaluates to nothing), which is probably why `/root` had been used in the first place. I'm going to revert those changes.

**Comment by AI-Pranto** [comment]
`$HOME` in docker actually points to the `\root` directory. My bad! @paulromano. 

LGTM @paulromano I have tested this on Docker and everything's fine.

**Review by wbinventor — APPROVED** [review]

**Comment by wbinventor** [comment]
Just tried this out in Docker for Windows and it works great!

**Review by pshriwise — APPROVED** [review]
I just tried this out as well and everything seems shipshape. Thanks @paulromano!

**Pull request merged**